perfkitbenchmarker/providers/rackspace/rackspace_network.py

```diff
@@ -13,140 +13,295 @@
-                updateport_cmd.extend(['--security-group', firewall])
-            updateport_cmd.append(port_id)
-
-            vm_util.IssueRetryableCommand(updateport_cmd, env=firewall_env)
-
-    def DisallowAllPorts(self):
-        """Closes all ports on the firewall."""
-        firewall_env = dict(os.environ.copy(),
-                            **util.GetDefaultRackspaceNeutronEnv(self))
-
-        for firewall in self.firewall_names:
-            firewall_cmd = []
-            firewall_cmd.extend([FLAGS.neutron_path,
-                                 'security-group-show',
-                                 '--format', 'value'])
-            firewall_cmd.append(firewall)
-
-            stdout, _ = vm_util.IssueRetryableCommand(firewall_cmd,
-                                                      env=firewall_env)
-
-            rules = [v for v in stdout.split('\n') if v != ''][2:-1]
-            for rule in rules:
-                rule_id = str(json.loads(rule)['id'])
-                rule_cmd = []
-                rule_cmd.extend([FLAGS.neutron_path,
-                                 'security-group-rule-delete'])
-                rule_cmd.append(rule_id)
-
-                vm_util.IssueRetryableCommand(rule_cmd, env=firewall_env)
-
-            firewall_cmd = [FLAGS.neutron_path]
-            firewall_cmd.extend(['security-group-delete'])
-            firewall_cmd.append(firewall)
-
-            vm_util.IssueRetryableCommand(firewall_cmd, env=firewall_env)
-
-            self.firewall_names.remove(firewall)
+class RackspaceSecurityGroup(resource.BaseResource):
+  """An object representing a Rackspace Security Group."""
+
+  def __init__(self, name):
+    super(RackspaceSecurityGroup, self).__init__()
+    self.name = name
+    self.id = None
+
+  def _Create(self):
+    cmd = util.RackCLICommand(self, 'networks', 'security-group', 'create')
+    cmd.flags['name'] = self.name
+    stdout, stderr, _ = cmd.Issue()
+    resp = json.loads(stdout)
+    self.id = resp['ID']
+
+  def _Delete(self):
+    if self.id is None:
+      return
+    cmd = util.RackCLICommand(self, 'networks', 'security-group', 'delete')
+    cmd.flags['id'] = self.id
+    cmd.Issue()
+
+  def _Exists(self):
+    if self.id is None:
+      return False
+    cmd = util.RackCLICommand(self, 'networks', 'security-group', 'get')
+    cmd.flags['id'] = self.id
+    stdout, stderr, _ = cmd.Issue()
+    return not stderr
+
+
+class RackspaceSecurityGroupRule(resource.BaseResource):
+  """An object representing a Security Group Rule."""
+
+  def __init__(self, sec_group_rule_name, sec_group_id, direction=None,
+               ip_ver=None, protocol=None, port_range_min=None,
+               port_range_max=None, source_cidr=None):
+    super(RackspaceSecurityGroupRule, self).__init__()
+    self.id = None
+    self.name = sec_group_rule_name
+    self.sec_group_id = sec_group_id
+    self.direction = direction if direction in SEC_GROUP_DIRECTIONS else INGRESS
```

Comment:
Yes, that's what I meant. My thought is that if someone (either you or
someone else) accidentally creates a security group with an invalid
configuration some time in the future, we'd rather see an error and solve
the problem rather than silently get wrong benchmarking results.

On Mon, Feb 15, 2016 at 10:53 AM, Carlos Torres <notifications@github.com>
wrote:

> In perfkitbenchmarker/providers/rackspace/rackspace_network.py
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/844#discussion_r52932780>
> :
>
> > +    cmd.flags['id'] = self.id
> > +    stdout, stderr, _ = cmd.Issue()
> > +    return not stderr
> > +
> > +
> > +class RackspaceSecurityGroupRule(resource.BaseResource):
> > +  """An object representing a Security Group Rule."""
> > +
> > +  def __init__(self, sec_group_rule_name, sec_group_id, direction=None,
> > +               ip_ver=None, protocol=None, port_range_min=None,
> > +               port_range_max=None, source_cidr=None):
> > +    super(RackspaceSecurityGroupRule, self).__init__()
> > +    self.id = None
> > +    self.name = sec_group_rule_name
> > +    self.sec_group_id = sec_group_id
> > +    self.direction = direction if direction in SEC_GROUP_DIRECTIONS else INGRESS
>
> or, do you mean it should raise an error for invalid values, instead of
> using a default?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/844/files#r52932780>
> .
>